### PR TITLE
[Emails] Add/track events for inbox upsells

### PR DESCRIPTION
**Pull request by AllTerrainDeveloper**
#### Changes proposed in this Pull Request

We need to track upsell events from mailbox screen. This PR adds the necessary tracking events for that.

During the development I found the the only way to pass some parameter to indicate that we are coming from inbox page, was to add a query parameter to the EmailsProviderComparison, as this component was handled by emails controller and not used actually as a component called by another. So ended up adding the context as a query parameter and receiving it in the controller.

I've took care of adding the events only in the case of everything is validated once added to the cart, so we wouldn't add false positive events.

#### Testing instructions

Go to inbox menu (shortcut: https://wordpress.com/inbox/{your-site-slug}) for the following cases:

**Site with no domain scenario**
User sees the "no domain" page that asks them to purchase a domain. This button should track a new event, under the key _calypso_inbox_upsell_ with _product='domain'_ and _context='inbox'_

**Site with no plan scenario**
User sees the "no domain" page that asks them to purchase a domain. This button should track a new event, under the key _calypso_inbox_upsell_ with _product='plan'_ and _context='inbox'_

**Site with no emails provider**
User sees the "no email subscriptions" page that asks them to purchase email for a domain, then this buttons should track a new event, under the key _calypso_email_providers_add_click_ with _product='email'_ and _context='inbox'_ and provider either _null_ (forward) _titan_ or _google_

**Adding new mailbox**
User clicks on the button to create a new mailbox from _**Upgrades -> Emails**_, then then when adding a new mailbox (continue button) a track event should be recorded  under the key _calypso_email_management_titan_add_mailboxes_continue_button_click_ or _calypso_email_management_gsuite_add_users_users_changed_ with _product='mailbox'_ and _context='inbox'_ and provider  _titan_ or _google_ (depending on Titan or Google mailbox)

**Adding new mailbox (variant 2)**
Do the same flow, but instead of starting from **_Upgrades -> Emails_** do it from inbox management screen (shortcut: https://wordpress.com/inbox/{your-site-slug}) and clicking on the bottom side button to add more mailboxes.

**Checking component view on Emails Comparison**
Check that the component view track event, now adds the _context='inbox'_ when necessary.

### REMARKS

Other event has changed the way they were logging the context, now we are adding "inbox" as context when needed, which might log context as:

- 'email-home-selected-domain'
- 'email-home-single-domain'
- 'email-purchase'
- 'inbox' 

Related to # {1200182182542585-as-1201064348369965}

### CODE REVIEW

File: client/my-sites/email/email-management/home/email-no-domain.jsx [hunk 1]
```diff
@@ -4,17 +4,26 @@ import PropTypes from 'prop-types';
...
 		hasDomainCredit( state, selectedSite.ID )
 	);
 
+	const trackEvent =
+		context != null
+			? () => {
+					recordInboxUpsellEvent( { product: 'domain', context } );
+			  }
+			: () => {};
+
```
Comment (AllTerrainDeveloper) [inline client/my-sites/email/email-management/home/email-no-domain.jsx]: Track only if context is from mailbox

File: client/my-sites/email/paths.js [hunk 2]
```diff
@@ -133,8 +133,13 @@ export function emailManagementForwarding( siteName, domainName, relativeTo = nu
  * @param {string} relativeTo - optional path prefix
  * @returns {string} the corresponding url
```
Comment (olaseni) [inline client/my-sites/email/paths.js:160]: Please include appropriate documentation for the new parameter

File: client/my-sites/email/paths.js [hunk 3]
```diff
@@ -133,8 +133,13 @@ export function emailManagementForwarding( siteName, domainName, relativeTo = nu
...
  */
-export function emailManagementPurchaseNewEmailAccount( siteName, domainName, relativeTo = null ) {
-	return emailManagementEdit( siteName, domainName, 'purchase', relativeTo );
+export function emailManagementPurchaseNewEmailAccount(
+	siteName,
+	domainName,
+	relativeTo = null,
+	urlParameters = null
```
Comment (olaseni) [inline client/my-sites/email/paths.js]: It might be better to use a `context`  or `sourceContext`parameter directly. The intent of the parameter is easier to discern that way.

```js
context : string = null
```

File: client/my-sites/email/controller.js [hunk 4]
```diff
@@ -75,6 +75,7 @@ export default {
 				<EmailProvidersComparison
 					comparisonContext="email-purchase"
 					selectedDomainName={ pageContext.params.domain }
+					context={ pageContext.query.context }
```
Comment (stephanethomas) [inline client/my-sites/email/controller.js]: I think we have a naming problem here. We're adding a new `context` prop but we already have this `comparisonContext` prop. It's not clear to me what's the difference between the two.

File: client/components/gsuite/gsuite-new-user-list/index.tsx [hunk 5]
```diff
@@ -57,6 +61,13 @@ const GSuiteNewUserList: FunctionComponent< Props > = ( {
 	};
 
 	const onUserAdd = () => {
+		if ( context === 'inbox-management' ) {
```
Comment (olaseni) [inline client/components/gsuite/gsuite-new-user-list/index.tsx]: It may be simpler to just use `inbox` as the context. Also since we are using this string all over the place, we should establish a constant for it to reduce mistakes and aid refactorings.

File: client/my-sites/email/email-management/home/email-no-domain.jsx [hunk 6]
```diff
@@ -4,17 +4,24 @@ import PropTypes from 'prop-types';
 import { useSelector } from 'react-redux';
 import Illustration from 'calypso/assets/images/customer-home/illustration--task-find-domain.svg';
 import EmptyContent from 'calypso/components/empty-content';
+import { recordInboxUpsellEvent } from 'calypso/my-sites/email/email-management/home/utils';
 import { hasDomainCredit } from 'calypso/state/sites/plans/selectors';
 
-const EmailNoDomain = ( { selectedSite, translate } ) => {
+const noop = () => {};
+
+const EmailNoDomain = ( { selectedSite, translate, context } ) => {
```
Comment (stephanethomas) [inline client/my-sites/email/email-management/home/email-no-domain.jsx]: Minor but we could keep the list of props ordered alphabetically.

File: client/my-sites/email/email-management/home/utils.js [hunk 7]
```diff
@@ -181,3 +181,16 @@ export function recordEmailAppLaunchEvent( { app, context, provider } ) {
 		provider,
 	} );
 }
+
+/**
+ * Tracks an event for the key 'calypso_inbox_upsell'.
+ *
+ * @param {app, context, provider} - app/context/provider should be string and must be provided.
+ */
+export function recordInboxUpsellEvent( { context, product, provider } ) {
```
Comment (stephanethomas) [inline client/my-sites/email/email-management/home/utils.js]: I'm wondering if we should make it more explicit that it's recording to Tracks, i.e. `recordInboxUpsellTracksEvent()`.

File: client/components/gsuite/gsuite-new-user-list/index.tsx [hunk 8]
```diff
@@ -57,6 +61,13 @@ const GSuiteNewUserList: FunctionComponent< Props > = ( {
...
+			recordInboxUpsellEvent( {
```
Comment (olaseni) [inline client/components/gsuite/gsuite-new-user-list/index.tsx]: BTW why are we recording an inbox upsell event here? This is just a change event that happens any time the users loads a new mailbox form. The user can still remove this mailbox without checking out.

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 9]
```diff
@@ -148,10 +153,17 @@ class EmailProvidersComparison extends Component {
 	};
 
 	goToEmailForwarding = () => {
-		const { currentRoute, selectedDomainName, selectedSite } = this.props;
+		const { context, currentRoute, selectedDomainName, selectedSite } = this.props;
 
 		recordTracksEvent( 'calypso_email_providers_add_click', { provider: 'email-forwarding' } );
 
+		if ( context === 'inbox-management' ) {
```
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: Already mentioned elsewhere, but reiterating. Let's shorten this to `inbox` and use a constant for it.

**Review by olaseni — CHANGES_REQUESTED** [review]
I think we are missing one scenario where the user has an email subscription but no mailboxes yet. Because this means that they have at least one mailbox that they don't have to purchase they are usually taken to a setup page rather checkout.

Can you track that down as well?

File: client/components/gsuite/gsuite-new-user-list/index.tsx (hunk 8 above)
Comment (AllTerrainDeveloper) [reply client/components/gsuite/gsuite-new-user-list/index.tsx]: In the Asan task this event should be recorded. Maybe I understood bad, but as per this text is what I understood:
> User clicks on the button to create a new mailbox

File: client/my-sites/email/controller.js (hunk 4 above)
Comment (AllTerrainDeveloper) [reply client/my-sites/email/controller.js]: I wondered the same @stephanethomas, my intention was not to change the original comparisonContext, as it's being used to track different events. So I tried to avoid regression by adding a new prop that would not interfere with other tracking methods. I am unsure if we should merge those two parameters... With something like

`pageContext.query.context ?? "email-purchase"`

File: client/my-sites/email/paths.js (hunk 3 above)
Comment (AllTerrainDeveloper) [reply client/my-sites/email/paths.js]: Good point

File: client/components/gsuite/gsuite-new-user-list/index.tsx (hunk 8 above)
Comment (AllTerrainDeveloper) [reply client/components/gsuite/gsuite-new-user-list/index.tsx]: I've understood everything correct, the Asana Card was not referring to this button :) You were right, thank you

File: client/my-sites/email/controller.js (hunk 4 above)
Comment (AllTerrainDeveloper) [reply client/my-sites/email/controller.js]: Finally I've merged both props.

File: client/my-sites/email/paths.js [hunk 10]
```diff
@@ -131,10 +171,22 @@ export function emailManagementForwarding( siteName, domainName, relativeTo = nu
  * @param {string} siteName - slug of the current site
  * @param {string} domainName - domain name of the account to add users to
  * @param {string} relativeTo - optional path prefix
+ * @param {string} context - optional path prefix
```
Comment (olaseni) [inline client/my-sites/email/paths.js]: Did you mean: `Optional context`? The doc is just a duplicate.

File: client/my-sites/email/paths.js [hunk 11]
```diff
@@ -105,13 +135,23 @@ export function emailManagementTitanControlPanelRedirect(
 	);
 }
 
-export function emailManagement( siteName, domainName, relativeTo = null ) {
+export function emailManagement( siteName, domainName, relativeTo = null, context = null ) {
 	let path;
+	let urlParam;
```
Comment (olaseni) [inline client/my-sites/email/paths.js]: You need a string default for this since `context` can be null sometimes.

File: client/my-sites/email/titan-add-mailboxes/index.jsx [hunk 12]
```diff
@@ -164,6 +168,15 @@ class TitanAddMailboxes extends Component {
 						// Stay on the page to show the relevant error
 						return;
 					}
+
+					if ( context === INBOX_MANAGEMENT ) {
+						recordInboxUpsellTracksEvent( {
+							context: context,
+							product: 'inbox',
```
Comment (olaseni) [inline client/my-sites/email/titan-add-mailboxes/index.jsx]: Nit: This doesn't seem right. What product is `inbox`? What do you intend to capture here?

File: client/my-sites/email/email-management/home/utils.js [hunk 13]
```diff
@@ -181,3 +181,16 @@ export function recordEmailAppLaunchEvent( { app, context, provider } ) {
...
 	} );
 }
+
+/**
+ * Tracks an event for the key 'calypso_inbox_upsell'.
+ *
+ * @param {app, context, provider} - app/context/provider should be string and must be provided.
+ */
+export function recordInboxUpsellTracksEvent( { context, product, provider } ) {
+	recordTracksEvent( 'calypso_inbox_upsell', {
```
Comment (olaseni) [inline client/my-sites/email/email-management/home/utils.js:191]: From use of this function elsewhere I can see that some parameters are not always provided. You can use null defaults?

File: client/my-sites/email/email-management/home/utils.js [hunk 14]
```diff
@@ -181,3 +181,16 @@ export function recordEmailAppLaunchEvent( { app, context, provider } ) {
...
+		context,
```
Comment (olaseni) [inline client/my-sites/email/email-management/home/utils.js:192]: The `context` parameter seems to be redundant since we always wrap the function call in:

```js
if ( context === INBOX_MANAGEMENT ) {
```

Do we expect it to be anything other than `inbox` ? If not let's remove it since the event name `calypso_inbox_upsell` already captures it.

File: client/my-sites/email/email-management/home/utils.js [hunk 15]
```diff
@@ -181,3 +181,16 @@ export function recordEmailAppLaunchEvent( { app, context, provider } ) {
...
+		product,
```
Comment (olaseni) [inline client/my-sites/email/email-management/home/utils.js]: Should this be maybe `section` ? It's looking more like a section from obervation.

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 16]
```diff
@@ -148,9 +153,19 @@ class EmailProvidersComparison extends Component {
...
-		const { currentRoute, selectedDomainName, selectedSite } = this.props;
+		const { comparisonContext, currentRoute, selectedDomainName, selectedSite } = this.props;
+
+		recordTracksEvent( 'calypso_email_providers_add_click', {
+			provider: 'email-forwarding',
+			context: comparisonContext,
+		} );
 
-		recordTracksEvent( 'calypso_email_providers_add_click', { provider: 'email-forwarding' } );
+		if ( comparisonContext === INBOX_MANAGEMENT ) {
```
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: This is redundant as all the information we need is already captured in previous call.

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 17]
```diff
@@ -148,9 +153,19 @@ class EmailProvidersComparison extends Component {
 	};
 
 	goToEmailForwarding = () => {
-		const { currentRoute, selectedDomainName, selectedSite } = this.props;
+		const { comparisonContext, currentRoute, selectedDomainName, selectedSite } = this.props;
+
+		recordTracksEvent( 'calypso_email_providers_add_click', {
+			provider: 'email-forwarding',
+			context: comparisonContext,
```
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: +1  :thumbsup:

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 18]
```diff
@@ -227,6 +242,15 @@ class EmailProvidersComparison extends Component {
 					// Stay on the page to show the relevant error(s)
 					return;
 				}
+
```
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx:248]: I think it's okay to have this right before we add to cart.

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 19]
```diff
@@ -284,6 +308,14 @@ class EmailProvidersComparison extends Component {
 					return;
 				}
 
+				if ( comparisonContext === INBOX_MANAGEMENT ) {
```
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: Same sentiments as above

File: client/my-sites/email/email-management/home/email-no-domain.jsx [hunk 20]
```diff
@@ -4,17 +4,32 @@ import PropTypes from 'prop-types';
...
-const EmailNoDomain = ( { selectedSite, translate } ) => {
+const noop = () => {};
+
+const EmailNoDomain = ( { context, selectedSite, translate } ) => {
 	const hasAvailableDomainCredit = useSelector( ( state ) =>
 		hasDomainCredit( state, selectedSite.ID )
 	);
 
-	if ( isFreePlan( selectedSite.plan.product_slug ) ) {
+	const isFreePlanProduct = isFreePlan( selectedSite.plan.product_slug );
```
Comment (olaseni) [inline client/my-sites/email/email-management/home/email-no-domain.jsx]: Let's make this a safe call.  

```js
 selectedSite?.plan?.product_slug ?? null
```

**Review by olaseni — COMMENTED** [review]
We are almost good to go. Just a few more changes.

File: client/my-sites/email/paths.js (hunk 11 above)
Comment (AllTerrainDeveloper) [reply client/my-sites/email/paths.js]: Good catch!

File: client/my-sites/email/titan-add-mailboxes/index.jsx (hunk 12 above)
Comment (AllTerrainDeveloper) [reply client/my-sites/email/titan-add-mailboxes/index.jsx]: @olaseni I think it's different to buy an email plan, than to add a mailbox to the plan. So with this we are differentiating when we buy a plan, from when we add a mailbox to an existing plan.

File: client/my-sites/email/paths.js [hunk 21]
```diff
@@ -131,10 +171,22 @@ export function emailManagementForwarding( siteName, domainName, relativeTo = nu
  * @param {string} siteName - slug of the current site
  * @param {string} domainName - domain name of the account to add users to
  * @param {string} relativeTo - optional path prefix
+ * @param {string} context - optional context prefix
```
Comment (olaseni) [inline client/my-sites/email/paths.js]: `context` is not a prefix. Please fix.

File: client/my-sites/email/controller.js [hunk 22]
```diff
@@ -73,7 +80,7 @@ export default {
 		pageContext.primary = (
 			<CalypsoShoppingCartProvider>
 				<EmailProvidersComparison
-					comparisonContext="email-purchase"
+					comparisonContext={ pageContext.query.context ?? 'email-purchase' }
```
Comment (olaseni) [inline client/my-sites/email/controller.js]: After looking at this again, it seems Dale's original intent for `comaprisonContext` may be quite different from what you want to do with `context`. From what I have learnt, yours is more like a way to track the referee?

IMHO the user could be redirected from`inbox` to make an `email-purchase`. Could combining their meanings trample the initial intent?

File: client/my-sites/email/titan-new-mailbox-list/index.jsx [hunk 23]
```diff
@@ -119,6 +119,7 @@ const TitanNewMailboxList = ( {
 
 TitanNewMailboxList.propTypes = {
 	children: PropTypes.node,
+	context: PropTypes.string,
```
Comment (olaseni) [inline client/my-sites/email/titan-new-mailbox-list/index.jsx]: This seems to be redundant. We are not using such a prop anywhere in this component.

File: client/my-sites/email/inbox/mailbox-selection-list/index.jsx [hunk 24]
```diff
@@ -121,7 +122,7 @@ const NewMailboxUpsell = () => {
 	const selectedSiteSlug = selectedSite?.slug;
 
 	const handleCreateNewMailboxClick = useCallback( () => {
-		page( `/email/${ selectedSiteSlug }` );
+		page( emailManagement( selectedSiteSlug, null, null, 'inbox' ) );
```
Comment (olaseni) [inline client/my-sites/email/inbox/mailbox-selection-list/index.jsx]: We can also use the constant here.

File: client/my-sites/email/email-management/home/email-list-inactive.jsx [hunk 25]
```diff
@@ -9,6 +9,7 @@ import { emailManagementPurchaseNewEmailAccount } from 'calypso/my-sites/email/p
 class EmailListInactive extends Component {
 	render() {
 		const {
+			context,
```
Comment (olaseni) [inline client/my-sites/email/email-management/home/email-list-inactive.jsx]: Please add proptype for this

File: client/my-sites/email/inbox/index.jsx [hunk 26]
```diff
@@ -18,6 +18,8 @@ import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
 import './style.scss';
 
+export const INBOX_MANAGEMENT = 'inbox';
```
Comment (olaseni) [inline client/my-sites/email/inbox/index.jsx]: This could be simply `INBOX`

File: client/my-sites/email/controller.js (hunk 22 above)
Comment (AllTerrainDeveloper) [reply client/my-sites/email/controller.js]: The use of this comparisonContext is to provide a Track context, in both cases either from inbox or from any other place. Any other place that uses this parameter, will keep recording everything as it was, but everything coming from inbox management will change the comparisonContext. I am happy to change it back to a new parameter called "context" but Stéphane wasn't happy with it.

Comment (stephanethomas) [reply client/my-sites/email/controller.js]: Following-up on what @olaseni mentioned, maybe we could indeed call it `referrer` instead?

Comment (AllTerrainDeveloper) [reply client/my-sites/email/controller.js]: After discussion in slack, we decided to split the meaning and add a second parameter called source.

Comment (stephanethomas) [reply client/my-sites/email/controller.js]: `source` is even better 👍.

File: client/my-sites/email/email-management/home/utils.js (hunk 15 above)
Comment (AllTerrainDeveloper) [reply client/my-sites/email/email-management/home/utils.js]: I think we want to track the upsell of a product, not of a section... Not sure

File: client/my-sites/email/inbox/index.jsx [hunk 27]
```diff
@@ -18,6 +18,8 @@ import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
 import './style.scss';
 
+export const INBOX = 'inbox';
```
Comment (stephanethomas) [inline client/my-sites/email/inbox/index.jsx:21]: I find it difficult to understand how we should use that constant, or what is it for (e.g. when I see this I wonder if `inbox` has not been localized on purpose). Maybe we could add a prefix to clarify that?

```suggestion
export const SOURCE_INBOX = 'inbox';
```

Should we also move it to a `constants.js` file, or do we prefer to keep it here?

File: client/my-sites/email/paths.js [hunk 28]
```diff
@@ -28,16 +28,24 @@ function resolveRootPath( relativeTo ) {
...
+	source = null
 ) {
 	if ( domainName ) {
-		return emailManagementEdit( siteName, domainName, productType + '/add-users', relativeTo );
+		return emailManagementEdit(
+			siteName,
+			domainName,
+			productType + '/add-users',
+			relativeTo,
+			source ? { source } : null
```
Comment (stephanethomas) [inline client/my-sites/email/paths.js]: Shouldn't we handle default values inside `emailManagementEdit()` rather than every time we call that method?

File: client/my-sites/email/paths.js [hunk 29]
```diff
@@ -105,13 +135,23 @@ export function emailManagementTitanControlPanelRedirect(
 	);
 }
 
-export function emailManagement( siteName, domainName, relativeTo = null ) {
+export function emailManagement( siteName, domainName, relativeTo = null, source = null ) {
 	let path;
+	let urlParam = '';
 
+	if ( source != null ) {
+		urlParam = `?source=${ source }`;
```
Comment (stephanethomas) [inline client/my-sites/email/paths.js]: Should we escape that `source` variable before injecting it in a path?

File: client/my-sites/email/paths.js [hunk 30]
```diff
@@ -105,13 +135,23 @@ export function emailManagementTitanControlPanelRedirect(
...
+		path = emailManagementEdit(
+			siteName,
+			domainName,
+			'manage',
+			relativeTo,
+			source ? { source } : null
+		);
 	} else if ( siteName ) {
-		path = '/email/' + siteName;
+		path = '/email/' + siteName + urlParam;
```
Comment (olaseni) [inline client/my-sites/email/paths.js]: Nit: Let's move `urlParam` into this block, close to where it's actually used. 

```js
const urlParam = source ? `?source=${ sourceEscaped }` : '';
path = '/email/' + siteName + buildQueryString( { source } );
```

See [PR](https://github.com/Automattic/wp-calypso/pull/56892) for `buildQueryString`

File: client/my-sites/email/paths.js (hunk 28 above)
Comment (olaseni) [reply client/my-sites/email/paths.js]: To speed things up a bit. Here's a [draft PR](https://github.com/Automattic/wp-calypso/pull/56892) that handles this: https://github.com/Automattic/wp-calypso/pull/56892

With that in place, even if the **variable** `source` is null, you could do:

```js
return emailManagementEdit( siteName, domainName, slug, relativeTo, { source } );
```

File: client/my-sites/email/email-management/home/email-no-domain.jsx [hunk 31]
```diff
@@ -4,17 +4,31 @@ import PropTypes from 'prop-types';
...
 	);
 
-	if ( isFreePlan( selectedSite.plan.product_slug ) ) {
+	const isFreePlanProduct = isFreePlan( selectedSite?.plan?.product_slug ?? null );
+
+	const trackEvent =
+		source != null
+			? () =>
+					recordInboxUpsellTracksEvent( {
+						product: isFreePlanProduct ? 'plan' : 'domain',
```
Comment (olaseni) [inline client/my-sites/email/email-management/home/email-no-domain.jsx]: These shouldn't be mutually exclusive. Also, a product is not interchangeable with a plan.

```js
freePlan: true,
product: domain,
```

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 32]
```diff
@@ -148,9 +153,24 @@ class EmailProvidersComparison extends Component {
...
+			comparisonContext,
+			currentRoute,
+			selectedDomainName,
+			selectedSite,
+			source,
+		} = this.props;
+
+		const eventProperties = {
+			provider: 'email-forwarding',
+			context: comparisonContext,
```
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: Why are we still sending the `comparisonContext`?

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 33]
```diff
@@ -148,9 +153,24 @@ class EmailProvidersComparison extends Component {
...
+		};
 
-		recordTracksEvent( 'calypso_email_providers_add_click', { provider: 'email-forwarding' } );
+		if ( source === INBOX ) {
+			eventProperties.product = 'forward';
```
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: This is redundant. The `provider` has already been set to `email-forwading` in the original event props. Let's avoid duplication.

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 34]
```diff
@@ -183,14 +203,22 @@ class EmailProvidersComparison extends Component {
...
 			mailboxes_valid: mailboxesAreValid ? 1 : 0,
 			provider: 'titan',
 			user_can_add_email: userCanAddEmail,
 			user_cannot_add_email_code: userCannotAddEmailReason ? userCannotAddEmailReason.code : '',
-		} );
+		};
+
+		if ( source === INBOX ) {
+			eventProperties.product = 'email';
+			eventProperties.provider = TITAN_PROVIDER_NAME;
```
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: This is a duplicate.

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 35]
```diff
@@ -183,14 +203,22 @@ class EmailProvidersComparison extends Component {
...
 			mailbox_count: validatedTitanMailboxes.length,
 			mailboxes_valid: mailboxesAreValid ? 1 : 0,
 			provider: 'titan',
 			user_can_add_email: userCanAddEmail,
 			user_cannot_add_email_code: userCannotAddEmailReason ? userCannotAddEmailReason.code : '',
-		} );
+		};
+
+		if ( source === INBOX ) {
+			eventProperties.product = 'email';
```
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: I still don't know how this can help. `eventProperties` already has a `comparisonContext`. If you think this is still needed, perhaps namespace it like so: `sourcePropert...`

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 36]
```diff
@@ -243,19 +272,26 @@ class EmailProvidersComparison extends Component {
...
 			mailbox_count: googleUsers.length,
 			mailboxes_valid: usersAreValid ? 1 : 0,
-			provider: 'google',
+			provider: GOOGLE_PROVIDER_NAME,
 			user_can_add_email: userCanAddEmail ? 1 : 0,
-		} );
+		};
+
+		if ( source === INBOX ) {
+			eventProperties.product = 'email';
```
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: Is `comparisonContext` enough here?

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 37]
```diff
@@ -390,6 +427,7 @@ class EmailProvidersComparison extends Component {
 						selectedDomainName={ selectedDomainName }
 						users={ googleUsers }
 						onReturnKeyPress={ this.onGoogleFormReturnKeyPress }
+						context={ comparisonContext }
```
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: Did you forget to remove this?

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 38]
```diff
@@ -493,6 +532,7 @@ class EmailProvidersComparison extends Component {
 				onReturnKeyPress={ this.onTitanFormReturnKeyPress }
 				showLabels={ true }
 				validatedMailboxUuids={ this.state.validatedTitanMailboxUuids }
+				context={ comparisonContext }
```
Comment (olaseni) [inline client/my-sites/email/email-providers-comparison/index.jsx]: Same. Did you forget?

File: client/my-sites/email/email-management/home/utils.js (hunk 15 above)
Comment (olaseni) [reply client/my-sites/email/email-management/home/utils.js]: A `product` already has some very definitive meanings in WPCOM, so this might cause some confusion.

Comment (AllTerrainDeveloper) [reply client/my-sites/email/email-management/home/utils.js]: I wanted to be consistent with the previous events recorded for Inbox

Comment (AllTerrainDeveloper) [reply client/my-sites/email/email-management/home/utils.js]: Everything clarified through Zoom call, Thanks!

File: client/my-sites/email/email-management/home/email-no-domain.jsx (hunk 31 above)
Comment (AllTerrainDeveloper) [reply client/my-sites/email/email-management/home/email-no-domain.jsx]: Clarified through Zoom call, Thanks!

**Review by olaseni — APPROVED** [review]
Looks Good.
